### ISSUE TITLE
DB stops

### ISSUE THREAD

**Issue by bytepoets-jhe** [simolus3/moor#3102]
Hi, thanks for reading!

Flutter 3.22.2, drift 2.19.1

We are having a problem with the database stopping in Flutter web.
I can reproduce it in Chrome of an old android tablet, it works fine on my macbook with chrome though.
Could not reproduce it yet on android or iOs apps

First thing here, WasmDatabase.open is missing the logStatement flag
So I extended the executor and logged the statements

It works initially, but after a lot of db requests it just freezes and does not return anything
I reduced the amount of db accesses from 200 to 36, but it did not get much better.
Mayvbe you got any ideas for that.

Now we would like to give the user feedback when this happens, because they are stuck in a loading state.
How yould I check that the db stopped working?

**Comment by simolus3** [comment]
Thank you for the report. I was unable to reproduce this myself thus far, but I have seen something similar in https://github.com/simolus3/drift/issues/3096 which indicates that it _may_ have been a regression introduced in drift 2.19.

In addition to the logs you've already collected, could you try running the app with the worker from https://storage.googleapis.com/simon-public-euw3/assets/drift/debug/drift_worker.js? That worker was compiled with debug flags and should post more information about this. Maybe that helps to reproduce this.

**Comment by bytepoets-jhe** [comment]
We already had the problem with drift 2.10.0, I updated to see if it helps

**Comment by bytepoets-jhe** [comment]
```
mapper: load sync status from db
main.dart.js:41769 Ensuring open
main.dart.js:41769 Running select: SELECT * FROM "evaluations"; with args: []
main.dart.js:41769 mapper: load sync status from db
main.dart.js:41769 Ensuring open
main.dart.js:41769 Running select: SELECT * FROM "evaluations"; with args: []
main.dart.js:41769 mapper: load sync status from db
main.dart.js:41769 Ensuring open
main.dart.js:41769 Running select: SELECT * FROM "evaluations"; with args: []
```
After a few db accesses it logs these but I never get anything from the db returned.
Switching the tab reloads the data from the db and I changed the tab here, that's why it tries to run the select more than once

**Comment by bytepoets-jhe** [comment]
Maybe this helps:
I have a branch where I can't reproduce it anymore, so I traced back my steps to see what fixes it.
I thought that maybe the sqlite3_flutter_libs update + new wasm file helped, but nope it only works when I add the debug wasm file from here: https://github.com/simolus3/sqlite3.dart/releases/tag/sqlite3_flutter_libs-0.5.22
Do you know what is the difference here?

**Comment by bytepoets-jhe** [comment]
@simolus3

**Comment by simolus3** [comment]
That's interesting! The debug version contains a lot of additional logs and was compiled with compiler optimizations disabled, but apart from that they should behave the same.

**Comment by bytepoets-jhe** [comment]
Here are the logs if I add
```
dependency_overrides:
  drift:
    git:
      url: https://github.com/simolus3/drift.git
      ref: wasm-logs
      path: drift
```

```
Ensuring open
main.dart.js:41827 Running select: SELECT * FROM "educator_answers" WHERE "evaluation_id" = ?; with args: [90e0de58-4bdf-4b9a-82df-6095928d3151]
main.dart.js:41827 [OUT]: Request (id = 194): StatementMethod.select: SELECT * FROM "educator_answers" WHERE "evaluation_id" = ?; with [90e0de58-4bdf-4b9a-82df-6095928d3151]
main.dart.js:41827 [IN]: SuccessResponse (id = 184): Instance of 'minified:G5'
main.dart.js:41827 mapper: load sync status from db
2main.dart.js:41827 Ensuring open
main.dart.js:41827 Running custom statement: UPDATE 'evaluations' SET remote_interaction_count = ?, synced_at = ? WHERE id = ? with args: [212, 2024-07-23T12:32:57.336363Z, 8dce761c-f3ba-4c90-9478-ad86fe7f00e9]
main.dart.js:41827 [OUT]: Request (id = 195): StatementMethod.custom: UPDATE 'evaluations' SET remote_interaction_count = ?, synced_at = ? WHERE id = ? with [212, 2024-07-23T12:32:57.336363Z, 8dce761c-f3ba-4c90-9478-ad86fe7f00e9]
main.dart.js:41827 Running select: SELECT * FROM "evaluations" WHERE "id" = ?; with args: [8dce761c-f3ba-4c90-9478-ad86fe7f00e9]
main.dart.js:41827 [OUT]: Request (id = 196): StatementMethod.select: SELECT * FROM "evaluations" WHERE "id" = ?; with [8dce761c-f3ba-4c90-9478-ad86fe7f00e9]
```

**Comment by simolus3** [comment]
From my current understanding, this appears to be the same as #3096 so I'll close this issue. If the fix from the #3096 doesn't work for you I'll have to take another look of course.
